### PR TITLE
Added drone and drone route table

### PR DESCRIPTION
**Pull request by ASLouw**
Also added corresponding endpoints for add, update and deactivate.

### CODE REVIEW

File: server/src/controllers/drone-route.controller.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { DroneRouteService } from '../services/drone-route.service';
+
+@Controller()
+export class DroneRouteController {
+  constructor(private readonly droneRouteService: DroneRouteService) {}
+
+  //add drone route info to drone route table in database
```
Comment (BCybercell) [inline server/src/controllers/drone-route.controller.ts:8]: Change to // add drone route info to drone route table in database

**Review by BCybercell — CHANGES_REQUESTED** [review]
Just a few small changes, commenting format

File: server/src/entity/drone-route.entity.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  id: number;
+
+  @ManyToOne(type => Drone, drone => drone.id)
+  drone: Drone;
+
+  @Column('text')
+  points: string;
+
+  @Column('text')
+  percentComplete: string;
```
Comment (EvansMatthew97) [inline server/src/entity/drone-route.entity.ts]: Use float

File: server/src/entity/drone.entity.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  avgSpeed: string;
+
+  @Column('text')
+  avgFlightTime: string;
+
+  @Column('text')
+  speed: string;
+
+  @Column('text')
+  flightTime: string;
```
Comment (EvansMatthew97) [inline server/src/entity/drone.entity.ts]: Use float

File: server/src/entity/drone.entity.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  id: number;
+
+  @Column('text')
+  name: string;
+
+  @Column('text')
+  avgSpeed: string;
+
+  @Column('text')
+  avgFlightTime: string;
```
Comment (EvansMatthew97) [inline server/src/entity/drone.entity.ts]: Use float

File: server/src/entity/drone.entity.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+@Entity()
+export class Drone {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column('text')
+  name: string;
+
+  @Column('text')
+  avgSpeed: string;
```
Comment (EvansMatthew97) [inline server/src/entity/drone.entity.ts]: Float

File: server/src/entity/drone.entity.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  @Column('text')
+  avgFlightTime: string;
+
+  @Column('text')
+  speed: string;
```
Comment (EvansMatthew97) [inline server/src/entity/drone.entity.ts]: Float

File: server/src/services/drone-route.service.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  //add drone route to Drone route table in database.
+  async addDroneRoute(id: number, points: string): Promise<boolean> {
+    const con = await this.databaseService.getConnection();
+    const droneRoute = new DroneRoute();
+
+    let droneIdfromDroneTable = await con
+      .getRepository(Drone)
+      .findOne({ id: id });
+
+    if ((await droneIdfromDroneTable) == undefined) {
```
Comment (EvansMatthew97) [inline server/src/services/drone-route.service.ts]: await unnecessary

**Review by EvansMatthew97 — CHANGES_REQUESTED** [review]

File: server/src/controllers/drone-route.controller.ts (hunk 1 above)
Comment (ASLouw) [reply server/src/controllers/drone-route.controller.ts:8]: done

File: server/src/entity/drone-route.entity.ts (hunk 2 above)
Comment (ASLouw) [reply server/src/entity/drone-route.entity.ts]: done

File: server/src/entity/drone.entity.ts (hunk 3 above)
Comment (ASLouw) [reply server/src/entity/drone.entity.ts]: done

File: server/src/entity/drone.entity.ts (hunk 4 above)
Comment (ASLouw) [reply server/src/entity/drone.entity.ts]: done

File: server/src/entity/drone.entity.ts (hunk 5 above)
Comment (ASLouw) [reply server/src/entity/drone.entity.ts]: done

File: server/src/entity/drone.entity.ts (hunk 6 above)
Comment (ASLouw) [reply server/src/entity/drone.entity.ts]: done

File: server/src/services/drone-route.service.ts (hunk 7 above)
Comment (ASLouw) [reply server/src/services/drone-route.service.ts]: done

**Review by EvansMatthew97 — APPROVED** [review]

**Review by BCybercell — APPROVED** [review]

**Pull request merged**